emulator/src/core/emulator/instructions/arithmetic.rs

```diff
@@ -271,5 +332,94 @@ mod tests {
 
         assert_eq!(e.reg['a'], 69-43);
     }
+    
+    #[test]
+    fn inx() {
+        let mut e = Emulator::new();
+
```

Comment:
As explained further down the line, these tests look very bloated to me. Usually blank lines indicate a set of logically coherent statements. So if there's multiple actions and different assertions for each maybe group the action and the assertion, otherwise the arrangements and assertions.